hydrus/data/crud.py

```diff
@@ -695,6 +695,74 @@ def delete_single(type_: str, session: scoped_session) -> None:
     return delete(instance.id, type_, session=session)
 
 
+def insert_modification_record(method: str, resource_url: str,
+                               session: scoped_session) -> str:
+    """
+    Insert a modification record into the database.
+    :param method: HTTP method type of related operation.
+    :param resource_url: URL of resource modified.
+    :param session: sqlalchemy session.
+    :return: ID of new modification record.
+    """
+    modification = Modification(method=method, resource_url=resource_url)
+    session.add(modification)
+    record_count = session.query(Modification).count()
+    if record_count > 1000:
+        oldest_record = session.query(Modification).order_by(Modification.timestamp.asc()).first()

```

Comment:
*Work in progress*

Reply:
While implementing multi-threading I was facing issues with in :memory: db of sqlite3. On some cursory look-up, I found out this https://docs.sqlalchemy.org/en/13/dialects/sqlite.html#using-a-memory-database-in-multiple-threads and made it work but still, there were unexpected bugs and use of `scopped_session` was not possible. So, instead of using in :memory: DB I used the database.db file as the default DB. The current approach with `socpped_session` is the recommended approach to add multi-threading, it works for all DBs except in :memory: Sqlite.